src/libsyntax/feature_gate.rs

```diff
@@ -1351,6 +1365,26 @@ pub const BUILTIN_ATTRIBUTES: &[BuiltinAttribute] = &[
                                                 "internal implementation detail",
                                                 cfg_fn!(rustc_attrs))),
 
+    (sym::rustc_allocator_nounwind, Whitelisted, template!(Word), Gated(Stability::Unstable,
+                                                sym::rustc_attrs,
+                                                "internal implementation detail",
+                                                cfg_fn!(rustc_attrs))),
+
+    (sym::rustc_builtin_macro, Whitelisted, template!(Word), Gated(Stability::Unstable,
+                                                sym::rustc_attrs,
+                                                "internal implementation detail",
+                                                cfg_fn!(rustc_attrs))),
+
+    (sym::rustc_promotable, Whitelisted, template!(Word), Gated(Stability::Unstable,
+                                                sym::rustc_attrs,
+                                                "internal implementation detail",
+                                                cfg_fn!(rustc_attrs))),
+
+    (sym::rustc_allow_const_fn_ptr, Whitelisted, template!(Word), Gated(Stability::Unstable,
+                                                sym::rustc_attrs,
+                                                "internal implementation detail",
+                                                cfg_fn!(rustc_attrs))),
```

Comment:
Oh, you whitelisted all the existing `rustc_*` attributes?

Reply:
Yep.
If an attribute is not defined/registered in any way, then it goes through the "`feature(custom_attribute)` fallback" (https://github.com/rust-lang/rust/issues/29642), something I want to remove from resolve (https://github.com/rust-lang/rust/pull/61660, https://github.com/rust-lang/rust/pull/57921).

Also, it's just useful for documentation.
https://github.com/rust-lang/rust/pull/62133 does the same thing.